### ISSUE TITLE
Role based access control for Cloud users

### ISSUE THREAD

**Issue by Diznik** [camunda-cloud/issues#29]
**Is your feature request related to a problem? Please describe.**
As a Camunda Cloud is used in large enterprises there are larger security and compliance concerns that mandate only the required access be granted to users. 

One customer in particular is running into an issue where a user account with a read-only option for Operate and a full Operate access (but not with client/cluster management), both at the cluster and global level, is imperative to integrating Camunda Cloud into their platform and use-cases. 

**Describe the solution you'd like**
The expectation is to apply a granularity of access by user - account level, role - job/responsibility and permissions - read/write/modify/delete for each application - Zeebe, Operate and Cloud Console. So that an Operate administrator can assign a user to a role with a set of permissions.

This granularity of access may also only apply to explicit resources/data (i.e. access process X but not process Y)

There should be both a UI and an API for creating  and modifying this authorization of access. 

Alignment here is expected - https://github.com/camunda/iam - for delivery.

**Your use case**
All large enterprises will require this functionality for automating their processes - especially those in heavily regulated industries

**Comment by ultraschuppi** [comment]
my 2ct:

* I believe we should totally know the different personas of our application and we should support them with meaningful roles that are attachable by org-admins to them ("developer", "operator" ...) instead of building a golden-hammer-role-permission-designer for our users
* org-roles are already implemented, permissions for those roles can be centralized (in accounts) or managed in each application
* this does not depend on IAM, we are using this in auth0 already. the payload needed for each application is just part of the issued JWT-token

**Comment by menski** [comment]
I agree that this is a important feature. Regarding the role permissions I would suggest with offering common roles, and wait for specific user feedback if we need an option for users to define their own roles, aka role designer.

What I see as a bigger challenge is the resource permissions, i.e. access process X but not process Y in Operate. My gut feeling would be that these restrictions are App specific, meaning they will not be specified in a global config. Although Operate, Tasklist and Optimize will all handle the same process instances, I would assume users which restrict access to process instances, also have different persons responsible for these Apps.
